spec/tests/002-replication-filter.md

```diff
@@ -0,0 +1,170 @@
+   * Add `airline_1x` and `airline_2x` in `inventory.airline`
+   * Update `airline_10`, `airline_22`, `airline_210` in `inventory.airline`
+   * Remove `route_10000`, `route_10001`, and `route_10010` in `inventory.route`
+6. Start the replicator with the same config as the step 2.
+7. Check that only docs specified in the documentIDs filters are replicated.
+
+## test_pull_document_ids_filter
+
+### Description
+
+Test that the replicator will pull only the docs specified in the document ids filter.
+
+### Steps
+
+1. Load `travel-sample` dataset into a database.
+2. Start a replicator: 
+    * collections : 
+      * `inventory.airport`
+         * documentIDs : `airport_1254`, `airport_1260`, `airport_1x`
+      * `inventory.landmark`
+         * documentIDs : `landmark_10019`, `landmark_10156`
+    * endpoint: `/travel-sample-inventory`
+    * type: pull
+    * continuos: false
+3. Wait until the replicator is stopped.
+4. Check that only docs specified in the documentIDs filters are replicated except `inventory.airport`.`airport_1x`.
+5. Update docs on SG
+   * Add `airport_1x` and `airport_2x` in `inventory.airport`
+   * Update `airport_1254`, `airport_1260`, `airport_1280` in `inventory.airport`
+   * Remove `landmark_10019`, `landmark_10156`, and `landmark_10622` in `inventory.landmark`
+6. Start the replicator with the same config as the step 2.
+7. Check that only docs specified in the documentIDs filters are replicated.
+
+## test_pull_channels_filter
+
+### Description
+
+Test that the replicator will pull only the docs specified in the channels filter.
+
+### Steps
+
+1. Load `travel-sample` dataset into a database.
+2. Start a replicator: 
+    * collections : 
+      * `inventory.airport`
+         * channels : `United States`, `France`
+      * `inventory.landmark`
+         * channels : `France`
+    * endpoint: `/travel-sample-inventory`
+    * type: pull
+    * continuos: false
+3. Wait until the replicator is stopped.
+4. Check that only docs in the specified channels are pulled.
+5. Update docs on SG
+   * Add `airport_1x` (United States), `airport_2x` (France) and `airport_3x` (United Kingdom) in `inventory.airport`
+   * Update `airport_3411` (United States), `airport_1254` (France), `airport_4346` (United Kingdom) in `inventory.airport`
+   * Remove `landmark_10144` (United States), `landmark_1006` (France) in `inventory.landmark`
+6. Start the replicator with the same config as the step 2.
+7. Check that only docs in the channels filters are replicated.
+
+## test_custom_push_filter
+
+### Description
+
+Test that the replicator will push only the docs that are passed from the push filter function.
+
+### Steps
+
+1. Load `travel-sample` dataset into a database.
+2. Start a replicator: 
+    * collections : 
+      * `inventory.airline`
+         * pushFilter:
+            * name: `documentIDs`
+            * params: `{ "documentIDs": { "inventory.airline": "airline_10", "airline_22", "airline_1x"} }`
```

Comment:
If this filter is specific to this collection, is it necessary to include the collection in the JSON?  Also the JSON is invalid.

Reply:
Not sure what I was thinking. It should be just an array of doc ids.